### ISSUE TITLE
Linux: Inconsistencies with XDG_CONFIG_HOME

### ISSUE THREAD

**Issue by micove** [PCSX2/pcsx2#381]
PCSX2 version: 1.3.0 GIT 9e8b0b5

This is an old issue that I recently remembered when I saw two pcsx2 directories in my .config folder.

Issue:
1) XDG_CONFIG_HOME=$HOME/.config
2) XDG_CONFIG_HOME="" <--- empty
30 unset XDG_CONFIG_HOME <--- not exists

All three in theory should use the same folder for the user configuration files.

The code actually uses:
1) $HOME/.config/PCSX2
2) $HOME/.config/pcsx2
3) $HOME/.config/pcsx2

Cause:
In pcsx2/gui/AppMain.cpp we have
        if( wxGetEnv(L"XDG_CONFIG_HOME", &xdg_home_value) ) {
            if ( xdg_home_value.IsEmpty() ) {
                // variable exist but it is empty. So use the default value
                user_local_dir = (wxDirName)Path::Combine( GetUserConfigDir() , wxDirName( L".config/pcsx2" ));
            } else {
                user_local_dir = (wxDirName)Path::Combine( xdg_home_value, pxGetAppName());
            }
        } else {
            // variable do not exist
            user_local_dir = (wxDirName)Path::Combine( GetUserConfigDir() , wxDirName( L".config/pcsx2" ));
        }

Basically we hardcode .config/pcsx2 twice and then use pxGetAppName() for case 1 which return the value set in:
./pcsx2/gui/AppInit.cpp:    SetAppName( L"PCSX2" );

Fix:
The code should probably be consistent and use all lower or all upper. Using  wxDirName(L"pcsx2") instead of pxGetAppName is an option as to not depend on the chance SetAppName gets changed.
